types/mongodb/index.d.ts

```diff
@@ -54,13 +54,64 @@ export class MongoClient extends EventEmitter {
     logout(options?: { dbName?: string }): Promise<any>;
     logout(options: { dbName?: string }, callback: MongoCallback<any>): void;
     /** http://mongodb.github.io/node-mongodb-native/3.0/api/MongoClient.html#startSession */
-    startSession(options?: any): ClientSession;
+    startSession(options?: SessionOptions): ClientSession;
 }
 
+/** https://docs.mongodb.com/manual/reference/method/Mongo.startSession/ */
+declare interface SessionOptions {
+    /** https://docs.mongodb.com/manual/core/read-isolation-consistency-recency/#causal-consistency */
+    causalConsistency?: boolean;
+    /**https://docs.mongodb.com/manual/reference/read-concern/#read-concern */
+    readConcern?: ReadConcern;
+    /** https://docs.mongodb.com/manual/reference/read-preference/#read-preference */
+    readPreference?: ReadPreference;
+    /**  */
+    retryWrites?: boolean;
+    /** https://docs.mongodb.com/manual/reference/write-concern/#write-concern */
+    writeConcern?: WriteConcern;
+}
+
+declare type ReadConcern = {
+    level?: "linearizable" /** https://docs.mongodb.com/manual/reference/read-concern-linearizable/#readconcern.%22linearizable%22 */
+            | "local" /** https://docs.mongodb.com/manual/reference/read-concern-local/#readconcern.%22local%22 */
+            | "majority" /** https://docs.mongodb.com/manual/reference/read-concern-majority/#readconcern.%22majority%22 */
+            | "snapshot"; /** https://docs.mongodb.com/manual/reference/read-concern-snapshot/#readconcern.%22snapshot%22 */
+    /** https://docs.mongodb.com/manual/reference/read-concern-linearizable/#readconcern.%22linearizable%22 */
+    maxTimeMS?: number;
+}
+
+/** https://docs.mongodb.com/manual/reference/write-concern/#write-concern */
+declare type WriteConcern = {
+    w?: number | "majority" ;
```

Comment:
Same as above regarding `type`.

Also it would be great if `CommonOptions ` extended this interface, instead of redefining `w`, `j` and `wtimeout`, so we avoid the duplication

Reply:
I can do that.